### ISSUE TITLE
can you add more numerical check?

### ISSUE THREAD

**Issue by dayekuaipao** [cuhkrlcourse/ierg6130-assignment#5]
This is a nice assignment based on PyTorch. I think it would be better to add more numerical check just like Staford CS231n does because sometimes rl code is hard to debug.  For example, in assignment2, my code seems to pass section 2.1 (it says your code should be bugfree), but cannot pass section 2.2. I debug my code in section 2.2 for a long time, then I find I just forget multiply learning rate in section 2.1. I think it`s a good idea to choose a certain random seed and make all the results reproducible, then we can add some numerical check to make debug process easier.

**Comment by pengzhenghao** [comment]
Good idea! Using a fixed seed can help us check whether the result is desired. Though I still has some concerns in practice: 

I can't guarantee the implementation of students is the same as mine, the result may slightly different though both implementations are right. What make thing worse is the different implementations may lead to greater and greater differences in the results as the codes goes from earlier section to later section. Personally, I found it's hard for me to design some qualitative criterions to judge student codes.

Second, sometimes you have to write the solution in some `TODO` codes blocks first before you can get the ground-truth result and compare it with students'. I don't think it's proper to embed the solution inside the assignment. So in that case the only thing I can do to check student codes is by a black-box test: I run a easier version of the algorithm (with small batch size and less training episodes). 

Do you have any suggestion in those cases?

**Comment by dayekuaipao** [comment]
Thanks for your reply! For the first concern, maybe there are some different opinions, I think it would be better to make the 
 results of different implementations the same. For example, in the section 2.2, the order of polynomial-basis can be different actually, but we may sort them so we can get the same results. For the second concern, I think if we can make the right output all the same, we can compare the right output with students` output, for example, the reward. I don`t know if it`s suitable.
